lib/src/workspace.rs

```diff
@@ -194,6 +207,17 @@ impl Workspace {
         workspace_root: &Path,
         git_repo_path: &Path,
     ) -> Result<(Self, Arc<ReadonlyRepo>), WorkspaceInitError> {
+        let mut git_repo_path = git_repo_path
+            .canonicalize()
+            .map_err(|_| WorkspaceInitError::PathNotFound(git_repo_path.to_path_buf()))?;
+        if !git_repo_path.ends_with(".git") {
+            git_repo_path.push(".git");
+
+            if !git_repo_path.exists() {
+                git_repo_path.pop();
+            }
+        }
```

Comment:
I think this should be done by CLI because it's basically a normalization of user input.

Reply:
Split out to #2830